tests/integration_tests/datasources/test_none.py

```diff
@@ -0,0 +1,64 @@
+"""NoCloud datasource integration tests."""
+import json
+
+import pytest
+from pycloudlib.lxd.instance import LXDInstance
+
+from tests.integration_tests.decorators import retry
+from tests.integration_tests.instances import IntegrationInstance
+from tests.integration_tests.integration_settings import PLATFORM
+from tests.integration_tests.util import verify_clean_log
+
+DS_NONE_BASE_CFG = """\
+datasource_list: [None]
+datasource:
+  None:
+    metadata:
+      instance-id: my-iid-uuid
+    userdata_raw: |
+      #cloud-config
+      runcmd:
+      - touch /var/tmp/success-with-datasource-none
+"""
+
+LXD_METADATA_DS_NONE_BASE_CFG = """\
+  /etc/cloud/cloud.cfg.d/99-force-dsnone.cfg:
+    when:
+    - create
+    - copy
+    create_only: false
+    template: dsnone.tpl
+    properties: {}
+"""
+
+
+@retry(tries=30, delay=1)
+def wait_for_cloud_init_status_file(instance: LXDInstance):
+    status_file = instance.read_from_file("/run/cloud-init/status.json")
+    assert len(status_file)
+
+
+@pytest.mark.lxd_use_exec
+@pytest.mark.skipif(
+    PLATFORM != "lxd_container",
+    reason="Requires LXD container with custom metadata setup",
```

Comment:
Is this still true? Can we remove this skip?

Reply:
yes we can drop this now that we are manipulating a running instance which already has networking.

Reply:
There is also a known bug with fallback network config on lxd_containers with virtual NICs https://bugs.launchpad.net/netplan/+bug/2022947 that requires that we provide network config that doesn't emit a "match:` clause by mac address due to netplan 0.106 behavior. For this case, we can inject the known working network `version: 2` configuration to make sure we can still ssh into lxd_containers across the DataSourceNone boundary (which generates fallback network config).